# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: extensions/permissions/PermissionDelegateHandler.cpp
# Commit: de69e3c1635a
# Full Hash: de69e3c1635a311b5b22716b65b93fc1899a37a4
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-08-01 09:48:10
# Regressor Bug: 1613431
# File Overlap Count: 1
# Description:
#   Bug 1613431 - Part 3: Handle synced setters return value. r=nika
#   
#   Depends on D83646
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83647
# ==============================================================================

diff -r b326d33f98cb -r de69e3c1635a extensions/permissions/PermissionDelegateHandler.cpp
--- a/extensions/permissions/PermissionDelegateHandler.cpp	Fri Jul 31 13:37:00 2020 +0000
+++ b/extensions/permissions/PermissionDelegateHandler.cpp	Fri Jul 31 13:37:13 2020 +0000
@@ -305,7 +305,7 @@
   }
 
   RefPtr<WindowContext> wc = mDocument->GetWindowContext();
-  NS_ENSURE_TRUE_VOID(wc);
+  NS_ENSURE_TRUE_VOID(wc && !wc->IsDiscarded());
 
   DelegatedPermissionList list;
   DelegatedPermissionList exactHostMatchList;
@@ -330,7 +330,7 @@
   WindowContext::Transaction txn;
   txn.SetDelegatedPermissions(list);
   txn.SetDelegatedExactHostMatchPermissions(exactHostMatchList);
-  txn.Commit(wc);
+  MOZ_ALWAYS_SUCCEEDS(txn.Commit(wc));
 }
 
 void PermissionDelegateHandler::UpdateDelegatedPermission(
@@ -374,7 +374,7 @@
 
   // We only commit if there is any change of permissions.
   if (changed) {
-    txn.Commit(wc);
+    MOZ_ALWAYS_SUCCEEDS(txn.Commit(wc));
   }
 }
 